core/modules/wbase/SendChannelShared.cc

```diff
@@ -0,0 +1,299 @@
+        if (thisTransmit == nullptr) {
+            throw Bug("_transmitLoop() _transmitQueue had nullptr!");
+        } else if (thisTransmit->result == nullptr) {
+            throw Bug("_transmitLoop() had nullptr result!");
+        }
+        idStr = makeIdStr(thisTransmit->result->queryid(), thisTransmit->result->jobid());
+
+        // Append the header for the next message to  thisTransmit->dataMsg.
+        auto sz = _transmitQueue.size();
+        // Is this really the last message for this SharedSendChannel.
+        bool reallyLast = (_lastRecvd && sz == 0);
+
+        proto::ProtoHeader* nextPHdr;
+        if (sz == 0) {
+            if (!_lastRecvd) {
+                throw Bug("SendChannelShared::_transmit() size=" + to_string(sz) + " but not last.");
+            }
+            // Create a header for an empty result using the protobuf arena from thisTransmit.
+            // This is the signal to the czar that this SharedSendChannel is finished.
+            nextPHdr = thisTransmit->createHeader();
+        } else {
+            auto nextT = _transmitQueue.front();
+            nextPHdr = nextT->header;
+        }
+        nextPHdr->set_endnodata(reallyLast);
+        string nextHeaderString;
+        nextPHdr->SerializeToString(&nextHeaderString);
+        thisTransmit->dataMsg += proto::ProtoHeaderWrap::wrap(nextHeaderString);
+        if (reallyLast) loop = false;
+        // Done with the queue for a while and this function may need to wait below.
+        qLock.unlock();
+        _qBackCv.notify_one();
+
+        // Limit the number of concurrent transmits.
+        bool scanInteractive = thisTransmit->scanInteractive;
+        // If there was an error, give this high priority.
+        if (thisTransmit->erred) {
+            scanInteractive = true;
+        }
+        bool largeResult = thisTransmit->largeResult;
+        int czarId = thisTransmit->czarId;
+        // Do NOT create transmitLock this with a mutex locked. High risk of deadlock.
+        wcontrol::TransmitLock transmitLock(*_transmitMgr, scanInteractive, largeResult, czarId);
+
+        // The first message needs to put its header data in metadata as there's
+        // no previous message it could attach its header to.
+        if (_firstTransmit.exchange(false)) {
+            // Put the header for the first message in metadata
+            // _metaDataBuf must remain valid until Finished() is called.
+            proto::ProtoHeader* thisPHdr = thisTransmit->header;
+            string thisHeaderString;
+            thisPHdr->SerializeToString(&thisHeaderString);
+            _metadataBuf = proto::ProtoHeaderWrap::wrap(thisHeaderString);
+            lock_guard<mutex> streamLock(streamMutex);
+            bool metaSet = _sendChannel->setMetadata(_metadataBuf.data(), _metadataBuf.size());
+            if (!metaSet) {
+                LOGS(_log, LOG_LVL_ERROR, "Failed to setMeta " << idStr);
```

Comment:
So, this really makes debugging difficult. The SSI SetMedata returns the reason for the failure and that appears to get lost. So, when this error occurs it next to impossible to figure out what went wrong. Could you pass through all related return codes so they actually get in a log message? This request applies to all SSI methods as I see other method calls also drop the reason for a failure.

Reply:
I'll try to fix that.